pkg/cloudprovider/suite_test.go

```diff
@@ -572,7 +572,12 @@ var _ = Describe("CloudProvider", func() {
 			ExpectApplied(ctx, env.Client, nodePool, nodeClass)
 			instanceTypes, err := cloudProvider.GetInstanceTypes(ctx, nodePool)
 			Expect(err).ToNot(HaveOccurred())
-			selectedInstanceType = instanceTypes[0]
+			selectedInstanceType, found := lo.Find(instanceTypes, func(it *corecloudproivder.InstanceType) bool {
```

Comment:
Why do we need this additional handling here? Feels odd that we would need to add additional logic to all test cases because we added filtering to `a1`. Maybe this speaks to the fact that we should statically select the instance type here. This might be necessary, hard for me to say, but on first look: It seems like a strange side-effect of these changes.

Reply:
This could be a static selection, it just so happened that the first instance type is now an a1 instance. IIRC this caused a failure when computing drift since `a1` instances are excluded from the `GetInstanceTypes` call for AL2023. I'm actually wondering if this could pose a problem for users upgrading, I'm not sure what happens if a NodeClass has an instance type option that is no longer surfaced by `GetInstanceTypes`. I'm not sure we need to worry about the existing node case since it wouldn't be working anyway.